metrics/utils.go

```diff
@@ -16,11 +16,39 @@ limitations under the License.
 
 package metrics
 
-import "strconv"
+import (
+	"strconv"
+
+	"go.opencensus.io/tag"
+)
 
 // ResponseCodeClass converts an HTTP response code to a string representing its response code class.
 // E.g., The response code class is "5xx" for response code 503.
 func ResponseCodeClass(responseCode int) string {
 	// Get the hundred digit of the response code and concatenate "xx".
 	return strconv.Itoa(responseCode/100) + "xx"
 }
+
+// MaybeInsertIntTag conditionally insert the tag when cond is true.
+func MaybeInsertIntTag(key tag.Key, value int, cond bool) tag.Mutator {
+	if cond {
+		return tag.Insert(key, strconv.Itoa(value))
```

Comment:
You may want to use `tag.Upsert` instead of `tag.Insert` as `insert` won't update existing key value pair. See https://godoc.org/go.opencensus.io/tag#Insert

Reply:
I think we want `Insert` here. This is just an utility function wrapping `tag.Insert`.

Reply:
I mean using `tag.Upsert` instead of `tag.Insert` in `source/source_stats_reporter.go`. Then this func will be `MaybeUpsertIntTag` and warpping `tag.Upsert`. Anyway this is something we can improve in separate PR.